### PR TITLE
Update documents.

### PR DESCRIPTION
**Pull request by junaruga**
I updated `README.md` and `example.yml`.

* Add explanation about ID, Macro (`macros` and `replaced_macros`) that are actually used in `ror.yml`.

* Add explanation about RPM List Builder

* Fix the parts of invalid format in the YAML file sample.
  Below ordered hash (?) looks invalid.
  I want to use `ror.yml` as standard that is actually used by RPM List Builder.

```
- name: Ruby 1.9.3
- requires: [v8314]
- packages:
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,8 +24,8 @@ An incomplete (in a way not all packages are mentioned here) example of python c
 # Recipe for python collections
 ---
 python33:
```
Comment (omron93) [inline README.md]: This is `ID`? Or what is ID intended to be?

**Review by omron93 — COMMENTED** [review]
Perl recipe (`perl.yaml`) uses `patch` clause (instead of `macros`). Is it supported? Should it be documented?

**Comment by junaruga** [comment]
@omron93 

RPM List Builder will not support the `patch` element.
Because I had considered and checked the `patch` parts in `perl.yaml`.
And I found that all the part was used to replace a macro variable.

And it can be replaced as `macros` or `replaced_macros` ideally in the future.
Though `patch` element should not be documented, it may be good idea to write current situation about `patch element.

I will work closely to replace those in `perl.yaml` with guys in charge of `perl.yaml` in the future.

For example, below part in `perl.yaml`.

```
    - perl516:
        install:
          - scl-utils-build
        patch: |
          --- a/macro-build
          +++ b/macro-build
          @@ -1,7 +1,7 @@
           %__perl %_scl_root/usr/bin/perl
           %__perl_provides /usr/lib/rpm/perl.prov.stack
           %__perl_requires /usr/lib/rpm/perl.req.stack
          -#%%perl_bootstrap 1
          +%perl_bootstrap 1
```

can be changed to

```
    - perl516:
        install:
          - scl-utils-build
        macros:
          perl_bootstrap: 1
```

or 

```
    - perl516:
        install:
          - scl-utils-build
        replaced_macros:
          perl_bootstrap: 1
```
depending on the context in the spec file.

or using proposed idea https://github.com/sclorg/rhscl-rebuild-recipes/issues/17 .

```
    - perl516:
        install:
          - scl-utils-build
        cmd: |
          patch -p1 spec_file <<EOF
--- a/macro-build
+++ b/macro-build
@@ -1,7 +1,7 @@
%__perl %_scl_root/usr/bin/perl
%__perl_provides /usr/lib/rpm/perl.prov.stack
%__perl_requires /usr/lib/rpm/perl.req.stack
-#%%perl_bootstrap 1
+%perl_bootstrap 1
EOF
```

File: README.md (hunk 1 above)
Comment (junaruga) [reply README.md]: Yes, the `python33` is ID. ID means Identity document [1]. Unique string.
Technically it is same with the key of the hash map in the YAML.
The ID manages `name`, `packages` element (= 1 SCL)
1 recipe file manages a list of ID.
I think that "Recipe ID"  would be better than "ID"

[1] https://en.wikipedia.org/wiki/ID

**Comment by junaruga** [comment]
Note: `requires` element in `perl.yaml`, `platforms` element in `example.yml` are still not supported by RPM List Builder.
Perhaps I may change the specification in the future, as a possibility.

File: README.md (hunk 1 above)
Comment (junaruga) [reply README.md]: I will update the document.

**Comment by omron93** [comment]
Thanks for explanation.

**Comment by junaruga** [comment]
@omron93 please wait a moment to merge. Now I am updating the document in local.

**Comment by junaruga** [comment]
@omron93 Finished to update it!
I have checked entire content of the document.

You can see modified README document here.
https://github.com/junaruga/rhscl-rebuild-recipes/blob/feature/update-readme-for-usage/README.md

**Comment by omron93** [comment]
It looks really good. Thanks.

**Comment by hhorak** [comment]
Looks good to me, merging.

**Comment by junaruga** [comment]
Thanks.

**Pull request merged**